File: user-management-service/User Management Service Command/src/main/java/de/fhdo/puls/user_management_service/command/configuration/DefaultTopicKafkaTemplate.java
package de.fhdo.puls.user_management_service.command.configuration;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.kafka.core.ProducerFactory;
import org.springframework.kafka.support.SendResult;
import org.springframework.util.concurrent.ListenableFuture;

public class DefaultTopicKafkaTemplate<K, V> extends KafkaTemplate<K, V> {
    @Value(value = "${kafka.message.topic.name}")
    private String defaultTopicName;

    public DefaultTopicKafkaTemplate(ProducerFactory<K, V> producerFactory) {
        super(producerFactory);
    }

    public DefaultTopicKafkaTemplate(ProducerFactory<K, V> producerFactory, boolean autoFlush) {
        super(producerFactory, autoFlush);
    }

    public ListenableFuture<SendResult<K, V>> send(V data) {
        return super.send(defaultTopicName, data);
    }
}

File: environment-service/Environment Service Command/src/main/kotlin/de/fhdo/puls/environment_service/command/configuration/DefaultTopicKafkaTemplate.kt
package de.fhdo.puls.environment_service.command.configuration

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration
import org.springframework.kafka.core.DefaultKafkaProducerFactory
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.kafka.core.ProducerFactory
import org.springframework.kafka.support.SendResult
import org.springframework.util.concurrent.ListenableFuture

@Configuration
class DefaultTopicKafkaTemplate<K, V> : KafkaTemplate<K, V> {
    @Value(value = "\${kafka.message.topic.name}")
    private val defaultTopicName: String? = null
    constructor(): super(DefaultKafkaProducerFactory<K,V>(HashMap<String, Any>()))
    constructor(producerFactory: ProducerFactory<K, V>) : super(producerFactory) {}
    constructor(producerFactory: ProducerFactory<K, V>, autoFlush: Boolean) : super(producerFactory!!, autoFlush) {}

    fun send(data: V): ListenableFuture<SendResult<K, V>> {
        return super.send(defaultTopicName!!, data)
    }
}